Sources/ClientRuntime/Networking/Http/CRT/CRTClientEngine.swift

```diff
@@ -29,22 +48,22 @@ public class CRTClientEngine: HttpClientEngine {
         }
 
         func getOrCreateConnectionPool(endpoint: Endpoint) throws -> HTTPClientConnectionManager {
-            guard let connectionPool = connectionPools[endpoint] else {
+            let poolID = ConnectionPoolID(endpoint: endpoint)
+            guard let connectionPool = connectionPools[poolID] else {
                 let newConnectionPool = try createConnectionPool(endpoint: endpoint)
-                connectionPools[endpoint] = newConnectionPool // save in dictionary
+                connectionPools[poolID] = newConnectionPool // save in dictionary
                 return newConnectionPool
             }
-
             return connectionPool
         }
 
         func getOrCreateHTTP2ConnectionPool(endpoint: Endpoint) throws -> HTTP2StreamManager {
-            guard let connectionPool = http2ConnectionPools[endpoint] else {
+            let poolID = ConnectionPoolID(endpoint: endpoint)
+            guard let connectionPool = http2ConnectionPools[poolID] else {
                 let newConnectionPool = try createHTTP2ConnectionPool(endpoint: endpoint)
-                http2ConnectionPools[endpoint] = newConnectionPool // save in dictionary
+                http2ConnectionPools[poolID] = newConnectionPool // save in dictionary
                 return newConnectionPool
             }
-
             return connectionPool
```

Comment:
`getOrCreateConnectionPool` and `getOrCreateHTTP2ConnectionPool` have a lot of duplicate code. Could we create a helper function that can be re-used in both that contains the majority of the logic?

Reply:
Discussed offline -- better to not refactor this part of the code and prioritize getting out the bug fix